Source/Core/DolphinQt2/GameList/GameFile.cpp

```diff
@@ -291,6 +294,62 @@ QString GameFile::GetLanguage(DiscIO::Language lang) const
   }
 }
 
+bool GameFile::IsInstalled() const
+{
+  if (m_platform != DiscIO::Platform::WII_WAD)
+    return false;
+
+  const std::string content_dir =
+      Common::GetTitleContentPath(m_title_id, Common::FromWhichRoot::FROM_CONFIGURED_ROOT);
+
+  if (!File::IsDirectory(content_dir))
+    return false;
+
+  // Since this isn't IOS and we only need a simple way to figure out if a title is installed,
+  // we make the (reasonable) assumption that having more than just the TMD in the content
+  // directory means that the title is installed.
+  const auto entries = File::ScanDirectoryTree(content_dir, false);
+  return std::any_of(entries.children.begin(), entries.children.end(),
+                     [](const auto& file) { return file.virtualName != "title.tmd"; });
+}
+
+bool GameFile::Install()
+{
+  if (m_platform != DiscIO::Platform::WII_WAD)
+    return false;
+
+  return DiscIO::CNANDContentManager::Access().Install_WiiWAD(m_path.toStdString());
+}
+
+bool GameFile::Uninstall()
+{
+  if (m_platform != DiscIO::Platform::WII_WAD)
+    return false;
+
+  return DiscIO::CNANDContentManager::Access().RemoveTitle(m_title_id,
+                                                           Common::FROM_CONFIGURED_ROOT);
+}
+
+bool GameFile::ExportWiiSaves()
+{
+  return CWiiSaveCrypted::ExportWiiSave(m_title_id);
+}
+
+QString GameFile::GetWiiFSPath() const
+{
+  if (m_platform == DiscIO::Platform::GAMECUBE_DISC)
+    return QStringLiteral("");
+
+  const std::string path =
+      StringFromFormat("%s/title/%08x/%08x/data/", File::GetUserPath(D_WIIROOT_IDX).c_str(),
+                       static_cast<u32>(m_title_id >> 32), static_cast<u32>(m_title_id));
+
+  if (!File::Exists(path))
```

Comment:
Is there any reason you are creating the title data directory here? When we get proper NAND support, this will not be possible to do here anymore, because the NAND should not be accessed at the same time by more than a single subsystem.

Regardless of any safety issue, I also disagree with this creation being here in GetWiiFSPath, which is supposed to only return the title data directory. And it looks like it may not even be needed, since GetTitleDataPath does that for you.